backend/src/main/java/harustudy/backend/content/service/ContentService.java

```diff
@@ -0,0 +1,165 @@
+package harustudy.backend.content.service;
+
+import harustudy.backend.auth.dto.AuthMember;
+import harustudy.backend.auth.exception.AuthorizationException;
+import harustudy.backend.content.domain.Content;
+import harustudy.backend.content.dto.ContentResponse;
+import harustudy.backend.content.dto.ContentsResponse;
+import harustudy.backend.content.dto.WritePlanRequest;
+import harustudy.backend.content.dto.WriteRetrospectRequest;
+import harustudy.backend.content.exception.ContentNotFoundException;
+import harustudy.backend.content.repository.ContentRepository;
+import harustudy.backend.member.domain.Member;
+import harustudy.backend.member.repository.MemberRepository;
+import harustudy.backend.participant.domain.Participant;
+import harustudy.backend.participant.domain.Step;
+import harustudy.backend.participant.exception.ParticipantNotFoundException;
+import harustudy.backend.participant.exception.StudyStepException;
+import harustudy.backend.participant.exception.ParticipantNotBelongToStudyException;
+import harustudy.backend.participant.repository.ParticipantRepository;
+import harustudy.backend.study.domain.Study;
+import harustudy.backend.study.exception.StudyNotFoundException;
+import harustudy.backend.study.repository.StudyRepository;
+import jakarta.annotation.Nullable;
+import java.util.List;
+import java.util.Objects;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@RequiredArgsConstructor
+@Transactional
+@Service
+public class ContentService {
+
+    private final StudyRepository studyRepository;
+    private final MemberRepository memberRepository;
+    private final ParticipantRepository participantRepository;
+    private final ContentRepository contentRepository;
+
+    @Transactional(readOnly = true)
+    public ContentsResponse findContentsWithFilter(
+            AuthMember authMember, Long studyId, Long participantId, @Nullable Integer cycle
+    ) {
+        List<Participant> participants = getParticipantIfAuthorized(
+                authMember, studyId);
+        Participant participant = filterSingleParticipantById(
+                participants, participantId);
+
+        List<Content> contents = participant.getContents();
+        if (Objects.isNull(cycle)) {
+            return getContentsResponseWithoutCycleFilter(contents);
+        }
+        return getContentsResponseWithCycleFilter(contents, cycle);
+    }
+
+    private List<Participant> getParticipantIfAuthorized(AuthMember authMember, Long studyId) {
```

Comment:
```suggestion
    private List<Participant> getParticipantsIfAuthorized(AuthMember authMember, Long studyId) {
```
participants s가 빠진것 같아요~